### ISSUE TITLE
`basePath` does not work with `urlMappingStrategy: "rewrite"` in `app` router

### ISSUE THREAD

**Issue by setowilliam** [QuiiBz/next-international#248]
**Describe the bug**
When using the `next-app` example, using `basePath` with `urlMappingStrategy: "rewrite"` redirects to `not-found` page when navigating to `http://localhost:3000/base`.

**To Reproduce**
1. Set `basePath` in `locales/client.ts` to `/base`
2. Set `basePath` in `next.config.js` to `/base`
3. Set `urlMappingStrategy` in `middleware.ts` to `rewrite`
4. Start dev server
5. Go to `http://localhost:3000/base`
6. `not-found` screen is shown

**Expected behavior**
From the description of `rewrite`:
> rewrite: https://example.com/products -> rewrite to https://example.com/en/products -> client doesn't see the locale in the url

I would expect that navigating to `http://localhost:3000/base` would rewrite to `http://localhost:3000/base/en`.

**Screenshots**

**About (please complete the following information):**
 - next-international `1.1.3`
 - Next.js `13.5.4`

**Comment by QuiiBz** [comment]
Good catch, we'll probably need to introduce a new `basePath` option to `createI18nMiddleware`. In the meantime, you could use the default `urlMappingStrategy`. Not ideal but I'll try to fix this bug asap.

**Comment by yukiyohure** [comment]
I came across the same Issue. I believe this library is one of the few libraries that meet the need for the Next.js app router users to "not want to change the url structure". So I hope this Issue will be fixed as soon as possible.

**Comment by yukiyohure** [comment]
@QuiiBz 
I came across the same Issue. I believe this library is one of the few libraries that meet the need for the Next.js app router users to "not want to change the url structure". So I hope this Issue will be fixed as soon as possible.
